android/src/main/java/io/ably/flutter/plugin/AblyMessageCodec.java

```diff
@@ -444,6 +450,13 @@ private Object decodeChannelMessageData(Map<String, Object> jsonMap) {
     return decodeMessageData(jsonMap.get(PlatformConstants.TxMessage.data));
   }
 
+  private MessageExtras decodeChannelMessageExtras(Map<String, Object> jsonMap) {
+    if (jsonMap == null) return null;
+    Object extras = jsonMap.get(PlatformConstants.TxMessageExtras.extras);
+    final Gson gson = new Gson();
+    return new MessageExtras(gson.fromJson(gson.toJson(extras, Map.class), JsonObject.class));
```

Comment:
I'm using that to convert a `Map` to `JsonObject` where `Map` need not only be a `Map<String, String>`

Reply:
So you're using `toJson` to remove non `String` keys. Is it just in case a user adds a non-string key on the dart-side? That's quite subtle, but also tackling the problem deeper than where it happens? Why not just maintain the invariant (extras don't have non-string keys) when a user creates them?

Reply:
1f0c17c1093de7f146b4b03af202a398a43150c8

Reply:
Improved readability: 1f0c17c1093de7f146b4b03af202a398a43150c8